### PR TITLE
[REVIEW] Port row_hasher to cudf++

### PR DESCRIPTION
**Pull request by trevorsm7**
Part of https://github.com/rapidsai/cudf/issues/2931 needed for `hash` and `hash_partition`

### CODE REVIEW

File: cpp/include/cudf/table/row_operators.cuh [hunk 1]
```diff
@@ -378,5 +379,102 @@ class row_lexicographic_comparator {
   order const* _column_order{};
 };  // class row_lexicographic_comparator
 
+/**---------------------------------------------------------------------------*
+ * @brief Computes the hash value of a row in the given column.
```
Comment (harrism) [inline cpp/include/cudf/table/row_operators.cuh]:
```suggestion
 * @brief Computes the hash value of an element in the given column.
```

**Comment by harrism** [comment]
@trevorsm7 There are no tests -- is this functionality sufficiently tested by hash and hash_partition tests?

**Comment by harrism** [comment]
rerun tests

**Comment by trevorsm7** [comment]
> @trevorsm7 There are no tests -- is this functionality sufficiently tested by hash and hash_partition tests?

Yes, `hash` and `hash_partition` should cover the entire surface of this, so thorough testing of those should fully cover row_hasher.

**Pull request merged**